### DISCUSSION TITLE
NaN-values in raster vs. contour plots

### DISCUSSION THREAD

**Discussion by Hvass-Labs** [observablehq/plot#1986] [Q&A] [answered]
This [Notebook](https://observablehq.com/d/f7b4814038c4292f) shows that NaN-values are transparent in raster-plots but color-filled in contour-plots. Is it possible to also make them transparent in contour-plots? The reason is that I would like to color the background green/red for pos/neg values, as we've discussed in another post here.

Thanks!

**Comment by Fil** [comment, marked as answer]
Since https://github.com/d3/d3-contour/releases/tag/v4.0.1 contours will go around -Infinity (but not NaN).

**Reply by Hvass-Labs** [reply]
Thanks! It works for a contour-plot when I set transparent values to `-Infinity` instead of `NaN` - but then it doesn't work for the raster-plot. It seems a bit inconsistent, but I'm sure you have your reasons for making it like this.

**Reply by Fil** [reply]
These are two different objects.

A contour represents a level set containing all points equal to the value *v*. If a point has a value -Infinity, a contour has to go around it. It's like an (infinitely deep) hole. If it's NaN, it means we don't know, so we don't assume it's high, low, or anything, and we ignore it when making the level set.

When we draw a raster, the logic is different: if the pixel has a value NaN, we don't know its value so we don't apply a color—it becomes transparent. If it has a value -Infinity, it is painted with the color that corresponds to that value (as given by the color scale). It happens that when you use the default scale turbo, everything below 0 is colored in black. (But if the color scale was defined in such a way that very low values were transparent, you would get a similar image.)

**Reply by Hvass-Labs** [reply]
OK, thanks for explaining.
